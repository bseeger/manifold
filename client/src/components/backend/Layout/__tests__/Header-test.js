import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../Header';
import build from 'test/fixtures/build';

describe("Backend.Layout.Header component", () => {

  const settings = build.entity.settings();
  const toggleOverlayMock = jest.fn();
  const togglerUserPanelMock = jest.fn();


  it('renders correctly', () => {
    const component = renderer.create(
      <Header
        authentication={{ authenticated: false }}
        notifications={{ notifications: {} }}
        location={{}}
        visibility={{ uiPanels: {} }}
        settings={settings}
        commonActions={{
          toggleSignInUpOverlay: toggleOverlayMock,
          toggleUserPanel: togglerUserPanelMock
        }}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
