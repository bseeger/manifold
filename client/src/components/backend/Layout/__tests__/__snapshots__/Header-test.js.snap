exports[`Backend.Layout.Header component renders correctly 1`] = `
<header
  className="header-app dark">
  <div
    className="header-container">
    <a
      className="logo"
      onClick={[Function]}
      style={Object {}}>
      <figure
        className="">
        <div>
          <i
            className="manicon manicon-manifold-logo" />
          <span
            className="screen-reader-text">
            Manifold Logo: Click to return to the browse page
          </span>
        </div>
      </figure>
    </a>
    <nav
      className="text-nav">
      <ul>
        <li
          className="">
          <a
            onClick={[Function]}
            style={Object {}}>
            Projects
          </a>
        </li>
        <li
          className="">
          <a
            onClick={[Function]}
            style={Object {}}>
            People
          </a>
        </li>
        <li
          className="">
          <a
            onClick={[Function]}
            style={Object {}}>
            Settings
          </a>
        </li>
      </ul>
    </nav>
    <nav
      className="menu-dropdowns">
      <ul>
        <li
          className="show-60">
          <a
            className="button-mode"
            onClick={[Function]}
            style={Object {}}>
            Exit Admin Mode
          </a>
        </li>
        <li>
          <button
            className="button-avatar"
            onClick={[Function]}>
            <span
              className="screen-reader-text">
              Click to login or open user settings
            </span>
            <figure
              className="avatar"
              style={undefined}>
              <i
                className="manicon manicon-person" />
            </figure>
          </button>
          <div
            className="panel-hidden">
            <nav
              className="user-menu menu-hidden">
              <i
                className="tail" />
              <ul>
                <li>
                  <a
                    href="#"
                    onClick={[Function]}>
                    <i
                      className="manicon manicon-person-pencil" />
                    Edit Profile
                  </a>
                </li>
                <li>
                  <button
                    onClick={[Function]}>
                    <i
                      className="manicon manicon-circle-arrow-out-right" />
                    Logout
                  </button>
                </li>
              </ul>
            </nav>
          </div>
        </li>
      </ul>
    </nav>
  </div>
  <div
    className="header-border" />
  <section
    className="header-notifications-container">
    <div
      className="header-notifications-list">
      <span />
    </div>
  </section>
</header>
`;
